### PR TITLE
Text block on step without title in rearrange modal [SCI-7195]

### PR DESCRIPTION
**Pull request by G-Chubinidze**
Jira ticket: [SCI-7195](https://scinote.atlassian.net/browse/SCI-7195)

### What was done
_improved step rearrange modal so elements have names_

[SCI-7195]: https://scinote.atlassian.net/browse/SCI-7195?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/vue/protocol/modals/reorderable_items_modal.vue [hunk 1]
```diff
@@ -24,7 +24,7 @@
                 <strong v-if="includeNumbers" class="step-element-number">{{ index + 1 }}</strong>
                 <i v-if="item.attributes.icon" class="fas" :class="item.attributes.icon"></i>
                 <span :title="item.attributes.name" v-if="item.attributes.name">{{ item.attributes.name }}</span>
-                <span v-else class="step-element-name-placeholder">{{ item.attributes.placeholder }}</span>
+                <span v-else :title="getTitle(item)">{{ getTitle(item) }}</span>
```
Comment (aignatov-bio) [inline app/javascript/vue/protocol/modals/reorderable_items_modal.vue:27]: We need a placeholder when the text block is empty. The name must be fixed in the serializer.

**Review by aignatov-bio — CHANGES_REQUESTED** [review]

File: app/javascript/vue/protocol/modals/reorderable_items_modal.vue (hunk 1 above)
Comment (artoscinote) [reply app/javascript/vue/protocol/modals/reorderable_items_modal.vue:27]: I think we should just leave placeholder as is, and move the getTitle(item) thing to the 'name' line above it. If, after removing images and tables, the title is empty, the placeholder will kick in.

File: app/javascript/vue/protocol/modals/reorderable_items_modal.vue [hunk 2]
```diff
@@ -24,7 +24,8 @@
                 <strong v-if="includeNumbers" class="step-element-number">{{ index + 1 }}</strong>
                 <i v-if="item.attributes.icon" class="fas" :class="item.attributes.icon"></i>
                 <span :title="item.attributes.name" v-if="item.attributes.name">{{ item.attributes.name }}</span>
-                <span v-else class="step-element-name-placeholder">{{ item.attributes.placeholder }}</span>
+                <span :title="getTitle(item)" v-else-if="item.attributes.name == ''">{{ getTitle(item) }}</span>
+                <span :title="item.attributes.placeholder" v-else class="step-element-name-placeholder">{{ item.attributes.placeholder }}</span>
```
Comment (artoscinote) [inline app/javascript/vue/protocol/modals/reorderable_items_modal.vue:28]: This should be done using an expression, not as if-else statements in the template, so something like:
`:title="item.attributes.name || getTitle(item) || item.attributes.placeholder"`

**Review by artoscinote — CHANGES_REQUESTED** [review]

File: app/javascript/vue/protocol/modals/reorderable_items_modal.vue (hunk 2 above)
Comment (artoscinote) [reply app/javascript/vue/protocol/modals/reorderable_items_modal.vue:28]: Maybe in a method called nameWithFallbacks, or something like that.

**Review by artoscinote — APPROVED** [review]

**Pull request merged**